src/builtins/core/date.rs

```diff
@@ -125,6 +125,44 @@ macro_rules! impl_with_fallback_method {
     };
 }
 
+/// Convenience methods for building a `PartialDate`
+impl PartialDate {
+    pub const fn with_era(mut self, era: Option<TinyAsciiStr<19>>) -> Self {
+        self.era = era;
+        self
+    }
+
+    pub const fn with_era_year(mut self, era_year: Option<i32>) -> Self {
+        self.era_year = era_year;
+        self
+    }
+
+    pub const fn with_year(mut self, year: Option<i32>) -> Self {
+        self.year = year;
+        self
+    }
+
+    pub const fn with_month(mut self, month: Option<u8>) -> Self {
+        self.month = month;
+        self
+    }
+
+    pub const fn with_month_code(mut self, month_code: Option<TinyAsciiStr<4>>) -> Self {
+        self.month_code = month_code;
+        self
+    }
+
+    pub const fn with_day(mut self, day: Option<u8>) -> Self {
+        self.day = day;
+        self
+    }
+
+    pub const fn with_calendar(mut self, calendar: Calendar) -> Self {
+        self.calendar = calendar;
+        self
+    }
+}
+
```

Comment:
Now that I think about this, can you add a const `new` constructor? Having all of these be `const` is useful, but they're less useful because we don't have a way to ergonomically create a const `PartialDate` without having to manually set all fields to `None`...

Reply:
I tried to add a `const` new as well. But it requires a `const` calendar constructor, which was ultimately the blocker. I'll play around with it a bit more.